go.mod

```diff
@@ -1,9 +1,10 @@
 module github.com/hashicorp/nomad-driver-podman
 
-go 1.21
+go 1.21.0
 
-replace (
+toolchain go1.21.1
```

Comment:
This seems a bit suspicious ... we should be able to leave the module with just `go1.21` and not force specific versions (i think?)